### ISSUE TITLE
Pass :x instead of block in proc leads to an error

### ISSUE THREAD

**Issue by vmeyet** [rubocop/rubocop#1485]
Opening an issue for the same reason as the comment of @monfresh at the end of that issue [Pass &:x? as an argument to lambda instead of a block.](https://github.com/bbatsov/rubocop/issues/1316).

here is the comment he made:

> I think this is also a bug for procs. When I have this:
> 
> ``` ruby
> validates :address,
>           presence: {
>             message: I18n.t('errors.messages.no_address')
>           },
>           unless: proc { |location| location.virtual? }
> rubocop complains and says:
> ```
> 
> Pass &:virtual? as an argument to proc instead of a block.
> If I change the code to this:
> 
> ``` ruby
> unless: proc(&:virtual?)
> Then I get Argument Error: no receiver during validation.
> ```

I got the very same issue:

``` ruby
# In a rails activemodel
after_save :geocode_and_add_city, :if => proc { |loc| loc.address_changed? }
```

which is flagged by rubocop (offense `Style/SymbolProc`)

```
app/models/location.rb:17:46: C: Pass &:address_changed? as an argument to proc instead of a block.
    after_save :geocode_and_add_city, :if => proc { |l| l.address_changed? }
```

but if changed to `proc(&:address_changed?)` correcting the offense then the code fails: `ArgumentError: no receiver given`. Basically it see the proc as taking no argument (especially not the instance) which led to the error.

This should probably be ignored by rubocop.
